### ISSUE TITLE
feedback

### ISSUE THREAD

**Issue by pavelsavara** [ilonatommy/reactWithDotnetOnWebWorker#1]
- [x] the application should link the github repo
- [x] the QR code doesn't change after you change the text
- [x] use https://aka.ms/dotnet-wasm-features to disable features to to make it smaller download
- [x] https://github.com/ilonatommy/reactWithDotnetOnWebWorker/blob/e6585bf18d360e326ee6e5f7403da0ed79f55599/dotnet/Program.cs#L16 could return `byte[]` and it would be `Uint8Array` on the other side. 
- [x] You can also send bytes in 3rd argument of `postMessage` as [transferable](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Transferable_objects)
- [x] nit: remove spinning react logo
- [x] nit: violet links on dark background are unreadable
- [x] is https://github.com/ilonatommy/reactWithDotnetOnWebWorker/blob/e6585bf18d360e326ee6e5f7403da0ed79f55599/react/src/setupProxy.js#L4 ever used ?

nice work on the ASCII diagrams!

@maraf please review as well

**Comment by ilonatommy** [comment]
> * [ ]  is https://github.com/ilonatommy/reactWithDotnetOnWebWorker/blob/e6585bf18d360e326ee6e5f7403da0ed79f55599/react/src/setupProxy.js#L4
>    ever used ?

Yes, when running the app locally. What about leaving the file commented out with an info to uncomment for running on the localhost?

**Comment by pavelsavara** [comment]
Yes, that would work.
Is that https://create-react-app.dev/docs/proxying-api-requests-in-development/ ?
I don't see `http-proxy-middleware` in package.json is it included by default ?

**Comment by ilonatommy** [comment]
It seems I tested it wrong (not on clean build). I cannot reproduce any failures for localhost serving, I'll keep it removed.
And yes, it was the mechanism from the link and the package should be required for that approach.

**Comment by maraf** [comment]
- [x] I would move these lines https://github.com/ilonatommy/reactWithDotnetOnWebWorker/blob/master/dotnet/main.js#L25-L28 to `QrImage.js`. Ideally get rid of id element or at least I would use some random generated value
- [x] It would be nice use Blazor's suggested way to load the image https://learn.microsoft.com/en-us/aspnet/core/blazor/images?view=aspnetcore-8.0#stream-image-data instead of base64
- [x] Maybe migrate .NET project to `Microsoft.NET.Sdk.WebAssembly`

Otherwise it looks great!

**Comment by pavelsavara** [comment]
this is not Blazor, Ilona also has Blazor demo in progress

yes, currently there is 
```
let uint8Array = new Uint8Array(e.data.image);
                let binaryString = Array.from(uint8Array).map(byte => String.fromCharCode(byte)).join('');
                let base64String = btoa(binaryString);
                document.getElementById("qrImage").src = `data:image/bmp;base64, ${base64String}`;
```

but we could possibly have just `createObjectURL`

```
const arrayBuffer = await imageStream.arrayBuffer();
    const blob = new Blob([arrayBuffer]);
    const url = URL.createObjectURL(blob);
    const image = document.getElementById(imageElementId);
```

**Comment by ilonatommy** [comment]
After the discussion offline I agree that we should stick to design: react only for frontend, WASM only for backend. My motivation of setting img source from WASM was purely to show that it's already possible from there but @maraf is right - it does not have any real value other than a fun-fact and won't be used that way.
Both feedbacks are fully applied now.
